src/components/ContinuationIn/UploadAffidavit.vue

```diff
@@ -0,0 +1,149 @@
+          :inputFile="business.affidavitFile"
+          :showErrors="getShowErrors"
+          :customErrorMessage.sync="customErrorMessage"
+          :isRequired="getShowErrors"
+          @fileValidity="onFileValidity($event)"
+          @fileSelected="onFileSelected($event)"
+        />
+      </v-col>
+    </v-row>
+  </v-expand-transition>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop } from 'vue-property-decorator'
+import { Getter } from 'pinia-class'
+import { useStore } from '@/store/store'
+import { StatusCodes } from 'http-status-codes'
+import { DateMixin, DocumentMixin } from '@/mixins'
+import { ExistingBusinessInfoIF, PresignedUrlIF } from '@/interfaces'
+import { PdfPageSize } from '@/enums'
+import FileUploadPreview from '@/components/common/FileUploadPreview.vue'
+@Component({
+  components: {
+    FileUploadPreview
+  }
+})
+export default class UploadAffidavit extends Mixins(DateMixin, DocumentMixin) {
+  readonly PdfPageSize = PdfPageSize
+
+  @Getter(useStore) getExistingBusinessInfo!: ExistingBusinessInfoIF
+  @Getter(useStore) getKeycloakGuid!: string
+  @Getter(useStore) getShowErrors!: boolean
+  @Getter(useStore) isContinuationInAffidavitRequired!: boolean
+
+  @Prop({ default: null }) readonly business!: ExistingBusinessInfoIF
+
+  // Local properties
+  customErrorMessage = ''
+  fileValidity = false
+
+  /**
+   * Called when FileUploadPreview tells us whether a file is valid.
+   * This is called right before the File Selected event.
+   */
+  onFileValidity (valid: boolean): void {
+    this.fileValidity = valid
+  }
+
+  /**
+   * Called when FileUploadPreview tells us about a new or cleared file.
+   * This is called right after the File Validity event.
+   */
+  async onFileSelected (file: File): Promise<void> {
+    if (file) {
+      if (this.fileValidity) {
+        // try to upload to Minio
+        let psu: PresignedUrlIF
+        try {
+          psu = await this.getPresignedUrl(file.name)
+          const res = await this.uploadToUrl(psu.preSignedUrl, file, psu.key, this.getKeycloakGuid)
+          if (!res || res.status !== StatusCodes.OK) throw new Error()
+        } catch {
+          // put file uploader into manual error mode by setting custom error message
+          this.customErrorMessage = this.UPLOAD_FAILED_MESSAGE
+          this.$forceUpdate() // force file upload component to react
+          return // don't add to list
+        }
+
+        // add properties reactively to business object and emit to parent coponent
+        this.$set(this.business, 'affidavitFile', {
+          name: file.name,
+          lastModified: file.lastModified,
+          size: file.size
+        } as File)
+        this.$set(this.business, 'affidavitFileKey', psu.key)
+        this.$set(this.business, 'affidavitFileName', file.name)
+        // emit the updated business object
+        this.$emit('updateBusiness', this.business)
```

Comment:
Use an `@Emit` decorator and separate method to emit please.

Reply:
Actually, you probably don't need to emit the updated business object back to the parent component.

Theoretically, the business object that you are updating here is the same object the parent is pointing to -- there is a single memory location to this object's memory and several variables point to it.

I say "theoretically" because you should test this, of course :)  But it would simplify your code. If this indeed works, please add a comment somewhere in this sub-component that the business is the same object that the parent is working with (not a copy -- the prop is a reference to the original object). I already added similar comments elsewhere in this code.

Reply:
Yes, the Prop works. Emit no longer needed

Reply:
Awesome. So does this pass-by-reference (multiple pointers to same object) make sense? It's a concept from long ago in C/C++ world.

Reply:
Got it! Make sense to me